proposed/0000-Unified-structured-data.md

```diff
@@ -0,0 +1,128 @@
+owners) only required when owners change 
+signature : mut Vec<Signature> // signs the `mut` fields above // 32 bytes (using e25519 sig)
+}
+```
+
+Fixed (immutable fields)
+- tag_type
+- identifier
+
+##Validation
+
+- To confirm name (storage location on network) we SHA512(tag_type + identifier). As these are much
+  smaller than the hash it prevents flooding of a network location.
+- To validate data we confirm signature using hash of (tag_type + version) as nonce. Initial `Put`
+  does not require this signature, but does require the owner contain a value.
+- If previous owners is set then the signature is confirmed using those keys (allows owenr
+  change/transfer etc.). In this case the previous owners is only required on first update and may be
+  remove in next version if the owners are not changed again.
+- To confirm sender of any `Put` (store or overwrite) then we check the signature of sender using same mechanism. For multiple senders we confirm at least 50% of owners have signed the request for `Put`
+
+When `Put` on the network this type is `StructuredData` with a subtype field. The network ignores this subtype
+except for collisions. No two data types with the same name and type can exist on the network.
+
+These types are stored in a disk based storage mechanism such as `Btree` at the NaeManagers (DataManagers) responsible for the area of the network at that `name`.
+
+These types will be limited to 100kB in size (as Immutable Chunks are also limited to 1Mb) which is for the time being a magic number.
+
+If a client requires these be larger than 100kB then the data component will contain a (optionally encrypted) datamap to be able to retrieve chunks of the network.
+
+The network will accept these types if `Put` by a Group and contains a message signed by at least 50% of owners as indicated. For avoidance of doubt 2 owners would require at least 1 have signed, 4 owners would require at least 2 etc. for majority control use an odd number of owners. Any `Put` must obey the mutability rules of these types.
+
+To update such a type the client will `Post` direct (not paying for this again) and the network will overwrite the existing data element if the request is signed by the owner and the version increments. To update a type then there must be an existing type of the same `Identity` and `type` whose owners (or optinally previous owners) includes at least a majority of this new type.
+
+For private data the data filed will be encrypted (at client discretion), for public data this need not be the case as anyone can read that, but only the owner can update it.
+
+##Client perspective 
+
+- Decide on a `type_tag` for a new type.
+- use whichever mechanism to create an `Identity` for this type
+- serialise any structure into `Vec<u8>` and include in data field (can be any structure that is serialisable)
+- store on network via `routing::Put(Identity: location, Data::StructuredData : data, u64: type_tag);`
```

Comment:
>So everybody starts by creating Structured Data (SD) and only if there is an overflow (beyond 100 KB) that one needs to turn to Immutable Data.

This RFC is only for structuredData, so if you have unstructured data like jpegs and files you would not look at this as an approach, you just use our existing self_encrypt methods. This proposal is for data that requires structure (more complex, more expensive than plain old immutable content). StrcuturedData i.e. create a database scheme, `Directory` types, `safecoin`, `login session`, `RDF / semantic` info and so on. Content is handled as efficiently as it can be with self encrypt and datamaps (i.e. `ImmutableData`.